### PR TITLE
fix(router): add parsing for `connector_request_reference_id` env

### PR DESCRIPTION
**Pull request by sai-harsha-vardhan**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
add parsing for `connector_request_reference_id` env

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Env
image

Config while running application
image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/utils.rs [hunk 1]
```diff
@@ -599,7 +599,7 @@ pub fn is_merchant_enabled_for_payment_id_as_connector_request_id(
     let config_map = &conf
         .connector_request_reference_id_config
         .merchant_ids_send_payment_id_as_connector_request_id;
-    config_map.contains(merchant_id)
+    config_map.iter().any(|id| id == merchant_id)
```
Comment (lsampras) [inline crates/router/src/core/utils.rs:602]: what's the difference?

File: crates/router/src/configs/settings.rs [hunk 2]
```diff
@@ -553,7 +553,7 @@ pub struct DelayedSessionConfig {
 
 #[derive(Debug, Deserialize, Clone, Default)]
 pub struct ConnectorRequestReferenceIdConfig {
-    pub merchant_ids_send_payment_id_as_connector_request_id: HashSet<String>,
+    pub merchant_ids_send_payment_id_as_connector_request_id: Vec<String>,
```
Comment (lsampras) [inline crates/router/src/configs/settings.rs:556]: why this?
Does config crate not support reading hashset from comma separated envs?

**Comment by lsampras** [comment]
> How did you test it?
>
> Compiler guided

you should be able to test this via running it locally with env vars since compiler won't point much here

File: crates/router/src/core/utils.rs (hunk 1 above)
Comment (sai-harsha-vardhan) [reply crates/router/src/core/utils.rs:602]: contains expects &String, so had to convert &str to &String. To avoid that, changed it to iter any logic

**Review by lsampras — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**